### PR TITLE
Chore/test for adding events

### PR DESCRIPTION
**Pull request by DerekMaffett**
Adds some basic testing for attaching events to data components and introduces some mock victory components we can reuse.

### CODE REVIEW

File: test/client/spec/mock-components.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  static displayName = "MockChart";
+  static role = "chart";
+
+  static defaultProps = {
+    dataComponent: <MockDataComponent/>,
+    labelComponent: <MockLabel/>,
+    groupComponent: <div/>
+  };
+
+  static getBaseProps = (props) => {
```
Comment (boygirl) [inline test/client/spec/mock-components.js:52]: getBaseProps should also return something in the "parent" namespace

File: test/client/spec/mock-components.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  render() {
+    const { text } = this.props;
+
+    return (
+      <p>`${text}`</p>
+    );
+  }
+}
+
+class MockChart extends React.Component {
```
Comment (boygirl) [inline test/client/spec/mock-components.js]: *nitpick* this should be renamed as it does not mock the functionality of VictoryChart. Maybe "MockVictoryComponent"?

File: test/client/spec/victory-util/add-events.spec.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import React from "react";
+import { forEach, get } from "lodash";
+import { mount } from "enzyme";
+import { addEvents } from "src/index";
+import { MockChart, MockLabel, MockDataComponent } from "../mock-components";
+
+describe("victory-util/add-events", () => {
+  const EventedMockChart = addEvents(MockChart);
+
+  const expectEventsTriggered = (scopeFn, testFn, expectations, wrapper) => {
```
Comment (boygirl) [inline test/client/spec/victory-util/add-events.spec.js]: I'm having a hard time following what is being asserted with this. We're using sinon / sinon-chai to spy on / write expectations about functions elsewhere. Could you use that here instead?

**Review by boygirl — CHANGES_REQUESTED** [review]
Please try using sinon. The other changes are nitpicks.

File: test/client/spec/victory-util/add-events.spec.js (hunk 3 above)
Comment (DerekMaffett) [reply test/client/spec/victory-util/add-events.spec.js]: Basically, this function is "Take a wrapper, get an array of components with `scopeFn`, map those components with `testFn`, and expect that each test result equals the corresponding expectation". I think I could rephrase this more as a pipeline if it would be more clear that way. Basically it just allows a comprehensive list of conditions to be tested on a set of components. 

`expect(scopeFn(wrapper).map(testFn)).to.eql(expectations)`

I've seen the examples of using ` eventHandlers: { onClick: spy }`, the pattern being to trigger an event and then check that `spy` was called with the signature `(evt, childProps, eventKey, this)`, with `childProps` and `eventKey` corresponding to the expected triggerer of the event. I think this could be a bit fragile though, since it doesn't test all the way through - at the end of the test, we only know that the handler function we supplied (normally to return a config object for the event) was invoked in some relation to the component we wanted to have an event on. We need to read the internals to know what that relation is though, and we still don't know whether anything changed in the app, or (I think) even what the target of the change would have been. 

Basically, `sinon.spy` tests are satisfied here: https://github.com/FormidableLabs/victory-core/blob/master/src/victory-util/events.js#L163

We could introduce a bug in the state change or even nuke this line of code and break the entire events system, and our current tests using `sinon.spy` would all pass since they're not able to see that far. https://github.com/FormidableLabs/victory-core/blob/master/src/victory-util/events.js#L166

That's why I opted to check the final result instead - it decouples our tests from internal details and gets full testing of a feature. 

Thoughts?

**Review by boygirl — APPROVED** [review]
That's much clearer. Thank you.
